packages/core-js/utils/ast/finders.js

```diff
@@ -9,42 +10,14 @@ function findContractDefinitions(astNode) {
   return Array.from(findAll('ContractDefinition', astNode));
 }
 
-/**
- * Get the given contract by id on the given AST
- * @param {number} contractId
- * @param {import("solidity-ast").SourceUnit|import("solidity-ast").SourceUnit[]} astNodes
- * @returns {import("solidity-ast").ContractDefinition}
- */
-function findContractNodeWithId(contractId, astNodes) {
-  if (Array.isArray(astNodes)) {
-    for (const astNode of astNodes) {
-      const contractDefiniton = findContractNodeWithId(contractId, astNode);
-      if (contractDefiniton) return contractDefiniton;
-    }
-  }
-
-  for (const contractDefiniton of findAll('ContractDefinition', astNodes)) {
-    if (contractDefiniton.id === contractId) {
-      return contractDefiniton;
-    }
-  }
-}
-
 /**
  * Get the given contract by name on the given AST
  * @param {string} contractName
- * @param {import("solidity-ast").SourceUnit|import("solidity-ast").SourceUnit[]} astNodes
+ * @param {import("solidity-ast").SourceUnit|import("solidity-ast").SourceUnit} sourceUnitNode
  * @returns {import("solidity-ast").ContractDefinition}
  */
-function findContractNodeWithName(contractName, astNodes) {
-  if (Array.isArray(astNodes)) {
-    for (const astNode of astNodes) {
-      const contractDefiniton = findContractNodeWithName(contractName, astNode);
-      if (contractDefiniton) return contractDefiniton;
-    }
-  }
-
-  for (const contractDefiniton of findAll('ContractDefinition', astNodes)) {
+function _findContractNodeWithName(contractName, sourceUnitNode) {
```

Comment:
This function is not exposed anymore as it is not reliable to be used because there could be contracts with the same name on different SourceUnits, but the user could not notice that and re-add the bug we had before.